src/Universum/Container/Class.hs

```diff
@@ -204,6 +204,84 @@ instance ToList (Set v)
 instance ToList (Seq a)
 instance ToList (Vector a)
 
+----------------------------------------------------------------------------
+-- ToPairs
+----------------------------------------------------------------------------
+
+{- | Type class for data types that can be converted to List of Pairs.
+ You can define 'ToPairs' by just defining 'toPairs' function.
+
+ But the following laws should be met:
+
+@
+'toPairs' m ≡ 'zip' ('keys' m) ('elems' m)
+'keys'      ≡ 'map' 'fst' . 'toPairs'
+'elems'     ≡ 'map' 'snd' . 'toPairs'
+@
+
+-}
+class ToPairs t where
+    {-# MINIMAL toPairs #-}
+    -- | Type of keys of the mapping.
+    type Key t :: *
+    -- | Type of value of the mapping.
+    type Val t :: *
+
+    -- | Converts the structure to the list of the key-value pairs.
+    --
+    -- >>> toPairs (HM.fromList [('a', "xxx"), ('b', "yyy")])
```

Comment:
I think it's better to add import, lie `import qualified Data.HashMap as HashMap` explicitly, because it's not clear what is `HM` from the context.